### PR TITLE
Feat!(schema): prohibit registering tables of different nesting

### PR DESCRIPTION
**Pull request by GeorgeSittas**
Plan to add another test to ensure the constructor enforces this logic too.

### CODE REVIEW

**Comment by GeorgeSittas** [comment]
Want to add another test to ensure the constructor enforces this logic too.

File: sqlglot/schema.py [hunk 1]
```diff
@@ -105,21 +106,21 @@ def __init__(
...
         )
         self._supported_table_args: t.Tuple[str, ...] = tuple()
 
     @property
     def empty(self) -> bool:
         return not self.mapping
 
-    def _depth(self) -> int:
+    def depth(self) -> int:
         return dict_depth(self.mapping)
```
Comment (tobymao) [inline sqlglot/schema.py:118]: can you cache the depth

Comment (GeorgeSittas) [reply sqlglot/schema.py:120]: done

File: sqlglot/schema.py [hunk 2]
```diff
@@ -305,12 +318,17 @@ def _normalize(self, schema: t.Dict) -> t.Dict:
         Returns:
             The normalized schema mapping.
         """
+        normalized_mapping: t.Dict = {}
         flattened_schema = flatten_schema(schema, depth=dict_depth(schema) - 1)
 
-        normalized_mapping: t.Dict = {}
+        depth = len(seq_get(flattened_schema, 0) or [])
```
Comment (tobymao) [inline sqlglot/schema.py:324]: just put this in the error message

Comment (GeorgeSittas) [reply sqlglot/schema.py:324]: Ah good point, will do

**Review by tobymao — APPROVED** [review]

**Pull request merged**